Library/Homebrew/cmd/pull.rb

```diff
@@ -320,6 +320,7 @@ def files_changed_in_patch(patchfile, tap)
   def current_versions_from_info_external(formula_name)
     versions = {}
     json = Utils.popen_read(HOMEBREW_BREW_FILE, "info", "--json=v1", formula_name)
+    json.force_encoding("UTF-8")
```

Comment:
Okay; I've [added a similar guard wrapper](https://github.com/apjanke/homebrew/blob/96b7d221867819baa2b19640a37f4f78c82e0b27/Library/Homebrew/cmd/pull.rb#L317-L325) so it only gets called for Rubys with string encoding information. What do you think now?